File: src/main/java/rbq/wtf/lycoris/client/utils/Logger.java
package rbq.wtf.lycoris.client.utils;

import rbq.wtf.lycoris.client.Client;

public class Logger {

    public static void info(String msg) {
        log(msg, LogLevel.INFO);
    }

    public static void info(String msg, String category) {
        log(msg, category, LogLevel.INFO);
    }

    public static void warning(String msg) {
        log(msg, LogLevel.WARNING);
    }

    public static void warning(String msg, String category) {
        log(msg, category, LogLevel.WARNING);
    }

    public static void error(String msg) {
        log(msg, LogLevel.ERROR);
    }

    public static void error(String msg, String category) {
        log(msg, category, LogLevel.ERROR);
    }

    public static void debug(String msg) {
        log(msg, LogLevel.DEBUG);
    }

    public static void debug(String msg, String category) {
        log(msg, category, LogLevel.DEBUG);
    }

    private static void log(String msg, LogLevel level) {
        if (Client.enabledLog && (Client.showDebugLevelLog || level != LogLevel.DEBUG)) {
            System.out.println("[Lycoris] [" + level.toString() + "] " + msg);
        }
    }

    private static void log(String msg, String category, LogLevel level) {
        if (Client.enabledLog && (Client.showDebugLevelLog || level != LogLevel.DEBUG)) {
            System.out.println("[Lycoris] [" + category + "] [" + level.toString() + "] " + msg);
        }
    }

    public enum LogLevel {
        INFO,
        WARNING,
        ERROR,
        DEBUG
    }
}

File: src/main/java/rbq/lycoris/client/utils/Logger.kt
package rbq.lycoris.client.utils

import rbq.lycoris.client.Client

object Logger {
    @JvmStatic
    fun info(msg: String) {
        log(msg, LogLevel.INFO)
    }

    @JvmStatic
    fun info(msg: String, category: String) {
        log(msg, category, LogLevel.INFO)
    }

    @JvmStatic
    fun warning(msg: String) {
        log(msg, LogLevel.WARNING)
    }

    @JvmStatic
    fun warning(msg: String, category: String) {
        log(msg, category, LogLevel.WARNING)
    }

    @JvmStatic
    fun error(msg: String) {
        log(msg, LogLevel.ERROR)
    }

    @JvmStatic
    fun error(msg: String, category: String) {
        log(msg, category, LogLevel.ERROR)
    }

    @JvmStatic
    fun debug(msg: String) {
        log(msg, LogLevel.DEBUG)
    }

    @JvmStatic
    fun debug(msg: String, category: String) {
        log(msg, category, LogLevel.DEBUG)
    }

    @JvmStatic
    private fun log(msg: String, level: LogLevel) {
        if (Client.enabledLog && (Client.showDebugLevelLog || level != LogLevel.DEBUG)) {
            println("[Lycoris] [$level] $msg")
        }
    }

    @JvmStatic
    private fun log(msg: String, category: String, level: LogLevel) {
        if (Client.enabledLog && (Client.showDebugLevelLog || level != LogLevel.DEBUG)) {
            println("[Lycoris] [$category] [$level] $msg")
        }
    }

    enum class LogLevel {
        INFO, WARNING, ERROR, DEBUG
    }
}